Makefile

```diff
@@ -342,12 +350,13 @@ clean-test: ## clean up debris from previously failed tests
 
 install: ## install binaries
 	@echo "$(WHALE) $@ $(BINARIES)"
-	@mkdir -p $(DESTDIR)/bin
-	@install $(BINARIES) $(DESTDIR)/bin
+	@install -d $(DESTDIR)/$(BINDIR) $(DESTDIR)/$(SBINDIR)
+	@install $(BINARIES_SYS) $(DESTDIR)/$(SBINDIR)
```

Comment:
Does the change from ` /usr/local/bin` to ` /usr/local/sbin` need changes in AppArmor or SELinux profiles to account for either location? I recall we hit that situation at some point when packaging docker.

Reply:
Yes, this change is highly breaking.

Reply:
> Does the change from ` /usr/local/bin` to ` /usr/local/sbin` need changes in AppArmor or SELinux profiles to account for either location? I recall we hit that situation at some point when packaging docker.

If you have an apparmor profile for the containerd itself (not talking about in-container stuff), than this needs to be fixed, of course. That's one of the primary duties of a distro packager (along w/ other distro specifc things like init scripts, install locations, hardening, etc)

Apropos packaging: are you talking about the stuff in docker-ce-packaging repo ?

This quite the opposite of how distro packaging works, since decades. The whole idea is generic workflows for generic tasks, so it easily integrates with other layers of the distro infrastructure - not doing very special magic for each individual package.
And the most vital part of such infrastructure are generic package build engines (eg. buildd, launchpad-builder, pbuilder, dck-buildpackage, etc) and their lower layers (like dpkg-buildpackage, etc), which just depend on the distro's standards being met. This docker-ce-packaging repo completely buypasses generic packaging infrastructures - it's a very special, proprietary things just for one application alone.

Reply:
By the way: I've meanwhile did a major rework, splitted it into several patches for being easier to review, more detailed description, etc ...

Haven't prepared a v3 queue yet, but you can find the patches in my debian queue:
(everything that's not prefixed "(debian)" is meant to be generic, only the debian patch is distro specific)

https://github.com/oss-qm/containerd --> branch: debian/maint-master

Reply:
It's not right to install containerd binary to sbin, as it can be executed by non privilege user, https://github.com/containerd/containerd/blob/master/docs/rootless.md

Reply:
> It's not right to install containerd binary to sbin, as it can be executed by non privilege user, https://github.com/containerd/containerd/blob/master/docs/rootless.md

The "s" stands for "system" not "superuser". There are lots of other daemons that can be run by unprivileged users (and sometimes even need to be), and also commands that can be used by unprivileged users under certain circumstances. The differenciation is not by "who can use it", but more kind of "does the average user do it".

Reply:
For rootless scenario, average user will do it.

Reply:
> For rootless scenario, average user will do it.

Will that "average user" call it directly or indirectly by some other frontend program ?

Reply:
Does it make difference if it's called directly or other programs? IMO, it's not appropriate for a program in usr/bin to call some other programs in usr/sbin.